src/basilisp/core.lpy

```diff
@@ -6854,17 +6864,13 @@
           (or (~field-kw-set ~key-gs)
               (contains? ~'_recmap ~key-gs)))
          (~'entry [~this-gs ~key-gs]
-          (cond
-            (contains? ~field-kw-set ~key-gs)
-            (map-entry ~key-gs (python/getattr ~this-gs (munge ~key-gs)))
-
-            (contains? ~'_recmap ~key-gs)
-            (map-entry ~key-gs (get ~'_recmap ~key-gs))))
```

Comment:
Rewriting `IAssociative.entry` in terms of `IAssocative.val_at` to prevent duplicated bugs.